h2/src/main/org/h2/mvstore/tx/TransactionMap.java

```diff
@@ -88,9 +88,9 @@ public long sizeAsLong() {
             for (int i = opentransactions.nextSetBit(0); i >= 0; i = opentransactions.nextSetBit(i+1)) {
                 MVMap<Long, Object[]> undoLog = store.undoLogs[i];
                 if (undoLog != null) {
-                    MVMap.RootReference rootReference = undoLog.getRoot();
+                    MVMap.RootReference rootReference = undoLog.flushAppendBuffer();
                     undoLogRootReferences[i] = rootReference;
-                    undoLogSize += rootReference.root.getTotalCount();
+                    undoLogSize += rootReference.root.getTotalCount() + rootReference.appendCounter;
```

Comment:
May be some similar change is needed in `sizeAsLongMax()` which should return an upper limit of size?

Reply:
I would rather amend MVMap.sizeAsLong() and MVMap.Empty(). 
Usage of append() / trimLast() meant to be a totally separate mode of operation, not to be combined with normal Map operations. I agree, that the whole design looks like a hacky backdoor, but since MVStore is homogeneous - everything is MVMap, we have to pretend that those objects (undo logs, temp buffers) are maps, although their usage pattern is quite different.